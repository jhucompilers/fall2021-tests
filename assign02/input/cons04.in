// build a list by applying f to each member of lst
function map(f, lst) {
  var result;
  if (nilp(lst)) {
    result = lst;
  } else {
    result = cons(f(car(lst)), map(f, cdr(lst)));
  }
  result;
}

function times2(x) {
  x * 2;
}

// result should be (6 8 10)
map(times2, list(3, 4, 5));
