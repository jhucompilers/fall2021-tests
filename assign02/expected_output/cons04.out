Result: (6 8 10)
